File: tools/bench/answers/python_haiku/s3/09_topwords.py
sentence = "the cat sat on the mat with the cat"
words = sentence.split()
freq = {}
for word in words:
    if word in freq:
        freq[word] += 1
    else:
        freq[word] = 1
sorted_words = sorted(freq.items(), key=lambda x: x[1], reverse=True)
for i in range(2):
    print(f"{sorted_words[i][0]} {sorted_words[i][1]}")

File: tools/bench/answers/go_haiku/s2/09_topwords.go
package main

import (
	"fmt"
	"sort"
	"strings"
)

func main() {
	sentence := "the cat sat on the mat with the cat"
	words := strings.Fields(sentence)
	freq := make(map[string]int)
	for _, w := range words {
		freq[w]++
	}

	type wordCount struct {
		word  string
		count int
	}
	var wc []wordCount
	for w, c := range freq {
		wc = append(wc, wordCount{w, c})
	}

	sort.Slice(wc, func(i, j int) bool {
		return wc[i].count > wc[j].count
	})

	for i := 0; i < 2 && i < len(wc); i++ {
		fmt.Printf("%s %d\n", wc[i].word, wc[i].count)
	}
}
